src/panoptes/utils/images/__init__.py

```diff
@@ -221,30 +222,56 @@ def _make_pretty_from_cr2(fname, title=None, **kwargs):
-        dtype (numpy.dtype, optional): The requested dtype for the new array.
+        saturation_level (scalar, optional): The saturation level. If not given then the
+            saturation level will be set to threshold times the maximum pixel value.
+        threshold (float, optional): The fraction of the maximum pixel value to use as
+            the saturation level, default 0.9.
+        bit_depth (astropy.units.Quantity or int, optional): The effective bit depth of the
+            data. If given the maximum pixel value will be assumed to be 2**bit_depth,
+            otherwise an attempt will be made to infer the maximum pixel value from the
+            data type of the data. If data is not an integer type the maximum pixel value
+            cannot be inferred and an IllegalValue exception will be raised.
+        dtype (numpy.dtype, optional): The requested dtype for the masked array. If not given
+            the dtype of the masked array will be same as data.
 
     Returns:
         numpy.ma.array: The masked numpy array.
+
+    Raises:
+        error.IllegalValue: Raised if bit_depth is an astropy.units.Quantity object but the
+            units are not compatible with either bits or bits/pixel.
+        error.IllegalValue: Raised if neither saturation level or bit_depth are given, and data
+            has a non integer data type.
     """
     if not saturation_level:
-        try:
-            # If data is an integer type use iinfo to compute machine limits.
-            dtype_info = np.iinfo(data.dtype)
-        except ValueError:
-            # Not an integer type. Assume for now we have 16 bit data.
-            saturation_level = threshold * (2 ** 16 - 1)
+        if bit_depth is not None:
+            try:
+                with suppress(AttributeError):
+                    bit_depth = bit_depth.to_value(unit=u.bit)
+            except u.UnitConversionError:
+                try:
+                    bit_depth = bit_depth.to_value(unit=u.bit / u.pixel)
+                except u.UnitConversionError:
+                    raise error.IllegalValue("bit_depth must have units of bits or bits/pixel, " +
+                                             f"got {bit_depth!r}")
+
+            bit_depth = int(bit_depth)
+            saturation_level = threshold * (2**bit_depth - 1)
         else:
-            # Data is an integer type, set saturation level at specified fraction of
-            # max value for the type.
-            saturation_level = threshold * dtype_info.max
+            # No bit depth specified, try to guess.
+            try:
+            # Try to use np.iinfo to compute machine limits. Will work for integer types.
+                saturation_level = threshold * np.iinfo(data.dtype).max
+            except ValueError:
+                # ValueError from np.iinfo means not an integer type.
+                raise error.IllegalValue("Neither saturation_level or bit_depth given, & data " +
```

Comment:
I'm just going to be a little anal about this. :) 
```suggestion
                raise error.IllegalValue("Neither saturation_level or bit_depth given, and data " +
```